### PR TITLE
juha/feature/267-add-attributes-pricing-component

### PR DESCRIPTION
**Pull request by juhaj77**
## 📄 **Pull Request Overview**
This pull request introduces the `AttributesPricing` component, which calculates the cost of upgrading levels. The maximum level is determined from statsPricingData, which is based on the currently available data.
### Key Features  
- **Level Range**:  
  The "From Level" and "To Level" range is dynamically set between the current value and the maximum value.  
- **Flexible Data Integration**:  
  The `statsPricingData` file supports rendering the cost table for the Hero Development Page.  
- **Rarity Class**:  
  The rarity class is required and can be one of the following numeric values: 1, 3, 5-8, 10. 
**Issue Number**: #267 

## 🔧 **Changes Made**

1. [Briefly describe changes you made]

2. [Any refactoring or clean-up tasks]

---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected.
- **JSDoc**: I have added or updated JSDoc comments for all relevant code.
- **Debugging**: No `console.log()` or other debugging statements are left.
- **Clean Code**: Removed commented-out or unnecessary code.
- **Tests**: Added new tests or updated existing ones for the changes made.
- **Documentation**: Documentation has been updated (if applicable).

---

## 📝 **Additional Information**
AttributesPricing

Provide any additional context or information that reviewers may need to know:

- **Screenshots**: [Include any screenshots or videos if the changes affect the UI]
- **Dependencies**: [Mention any new dependencies or breaking changes]
- **Known Issues**: [List any known issues or limitations]

### CODE REVIEW

**Review by leolabdev — CHANGES_REQUESTED** [review]
Hi, I would like your object to be able to process some class, for example HeroStatsPriceManager, since React should be free from such logic. I slightly transferred your solution so that there was no useEffect, and instead of useState I would use useReducer in this case.

Also in your solution there is a problem that when you change the characteristic, the initial minimum levels are somehow strangely bound, for example for speed the minimum level is 5 instead of 1, for impact force 4, for size 3... etc

image

**Comment by leolabdev** [comment]
And @Jonroi and I would like to commend you, you came up with a very clever solution with the price calculation.

**Comment by juhaj77** [comment]
> Also in your solution there is a problem that when you change the characteristic, the initial minimum levels are somehow strangely bound, for example for speed the minimum level is 5 instead of 1, for impact force 4, for size 3... etc

This is about the default level. For example, if speed is already at level 5, the level range starts from there, so it doesn't need to be selected separately. It varies based on the character data.

File: frontend-next-migration/src/entities/Hero/model/stats/AttributesPricingHelper.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { statsPricingData } from './statsPricingData';
+import { SetStateAction } from 'react';
+import { Stat } from '../../ui/AttributesPricing3/AttributesPricingV3';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Hero/model/stats/AttributesPricingHelper.tsx:3]: move this one to types folder

**Review by leolabdev — CHANGES_REQUESTED** [review]

**Review by leolabdev — APPROVED** [review]

**Pull request merged**